adafruit_led_animation/animation.py

```
@@ -0,0 +1,507 @@
+
+**Software and Dependencies:**
+
+* Adafruit CircuitPython firmware for the supported boards:
+  https://circuitpython.org/downloads
+
+"""
+
+try:
+    from time import monotonic_ns
+except ImportError:
+    import time
+
+    def monotonic_ns():
+        """
+        Implementation of monotonic_ns for platforms without time.monotonic_ns
+        """
+        return int(time.time() * 1000000000)
+
+import random
+from .color import BLACK, RAINBOW
+
+__version__ = "0.0.0-auto.0"
+__repo__ = "https://github.com/adafruit/Adafruit_CircuitPython_LED_Animation.git"
+
+
+class Animation:
+    """
+    Base class for animations.
+    """
+    # pylint: disable=too-many-arguments
+    def __init__(self, pixel_object, speed, color, peers=None, paused=False):
+        self.pixel_object = pixel_object
+        self._speed_ns = 0
+        self.speed = speed  # sets _speed_ns
+        self._color = color
+        self._next_update = monotonic_ns()
+        self.pixel_object.auto_write = False
+        self.color = color
+        self.peers = peers if peers else []
+        self._paused = paused
+        self._time_left_at_pause = 0
+
+    def animate(self):
+        """
+        Call animate() from your code's main loop.  It will draw the animation draw() at intervals
+        configured by the speed property (set from init).
+
+        :return: True if the animation draw cycle was triggered, otherwise False.
+        """
+        if self._paused:
+            return False
+
+        now = monotonic_ns()
+        if now < self._next_update:
+            return False
+
+        self.draw()
+        if self.peers:
+            for peer in self.peers:
+                peer.draw()
+        self.show()
+        for peer in self.peers:
+            peer.show()
+
+        self._next_update = now + self._speed_ns
+        return True
+
+    def draw(self):
+        """
+        Animation subclasses must implement draw() to render the animation sequence.
+        """
```

add a `raise NotImplementedError()` here instead of leaving it empty

---

Good catch.